src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ConnectSqlCmdCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Management.Common;
+using System;
+using System.Data;
+using System.Data.Common;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    public class ConnectSqlCmdCommand : SqlCmdCommand
+    {
+        internal ConnectSqlCmdCommand(string server, string username, string password) : base(LexerTokenType.Connect)
+        {
+            Server = server;
+            UserName = username;
+            Password = password;
+        }
+
+        public string Server { get; private set; }
+        public string UserName { get; private set; }
+        public string Password { get; private set; }
+
+
+        /// <summary>
+        /// attempts to establish connection with given params
+        /// </summary>
+        /// <returns>returns the connection object is successful, throws otherwise</returns>
+        public DbConnection Connect()
+        {
+            //create SqlConnectionInfo object
+            SqlConnectionInfo connectionInfo = new SqlConnectionInfo();
+            if (Server != null && Server.Length > 0)
+            {
+                connectionInfo.ServerName = Server;
+            }
+            if (UserName != null && UserName.Length > 0)
+            {
+                connectionInfo.UseIntegratedSecurity = false;
+                connectionInfo.UserName = UserName;
+                connectionInfo.Password = Password;
+            }
+            else
+            {
+                connectionInfo.UseIntegratedSecurity = true;
+            }
+
+            DbConnection dbConnection = AttemptToEstablishCurConnection(connectionInfo);
+            return dbConnection;
+        }
+
+        /// <summary>
+        /// called when we need to establish new connection for batch executio as a
+        /// result of "connect" command processing
+        /// </summary>
+        /// <param name="ci"></param>
+        /// <returns></returns>
+        private DbConnection AttemptToEstablishCurConnection(SqlConnectionInfo ci)
```

Comment:
Can we use `ConnectionService::OpenSqlConnection` to open the SqlConnection?  Ideally the iteractions with SqlClient to open new connections are consolidated there to centralize that logic.

Reply:
So currently two things
1. The Batchparse --> Service layer reverse ref - but this can be solved by moving the OpenSqlConnection to ConnectionServiceCore
2. The second is a little weird that we need ConnectionInfo object for OpenSqlConnection which is owner URI based and is cached (it seems) but here we only want temporary connection for the length of batch execution and the overall query connection should remain same.

I will keep in mind for later in refactoring can help here - but for now will leave this one